src/Tasks/ManifestUtil/Manifest.cs

```diff
@@ -485,6 +498,21 @@ private static void UpdateFileReference(BaseReference f, string targetFrameworkV
             }
             f.Hash = hash;
             f.Size = size;
+
+            //
+            // .NETCore Launcher.exe based Deployment: If the filereference is for apphost.exe, we need to change
+            // the ResolvedPath and TargetPath to {assemblyname}.exe before we write the manifest, so that the
+            // manifest does not have a file reference to apphost.exe
+            //
+            string fileName = Path.GetFileName(f.ResolvedPath);
+            if (LauncherBasedDeployment &&
+                fileName.Equals(Constants.AppHostExe, StringComparison.InvariantCultureIgnoreCase) &&
+                !String.IsNullOrEmpty(AssemblyName))
+            {
+                f.ResolvedPath = Path.Combine(Path.GetDirectoryName(f.ResolvedPath), AssemblyName);
+                f.TargetPath = BaseReference.GetDefaultTargetPath(f.ResolvedPath);
+            }
```

Comment:
nit:
Turn the below if into an else if? It could also be simplified a little with a ternary operator, although some would call that messier.

Reply:
Since I am looking to get this checked in at the earliest and reduce the churn after testing, I will make this change next time.